touchup(drop down menu): applied icons and background colors for UX

rdubbi opened the pull request:
Closes #629

### Description of changes
All changes were made in AddNodeMenuItems.tsx. I imported useTheme. I built upon shownNodeTypes.map so that it carried over the icons from node.ts and the background colors from theme.ts.

![Screen Shot 2025-01-07 at 8 45 38 PM](https://github.com/user-attachments/assets/76393cb1-0399-40dd-b0ca-5ee5c4b1d46f)

-

### Additional context

General "New Guy, please help" Disclaimer: 
I graduated from a coding bootcamp and have spent months applying to jobs. I thought contributing to some open source projects would be a great opportunity to grow as a software engineer, make the world a better place, and bolster my resume. This is my very first open source contribution, and my very first time using TypeScript. All of this to say, I tried my best to follow all of the contributor guidelines and "clean code" best practices, so if there are any glaring mistakes or deviations from contributor best practices, I promise it is ignorance, not sloppiness, and I wholeheartedly welcome any and all feedback!

Future Changes: 
From the directions in issue 629, it seemed to me that the change was only to be applied to the "Add Node" drop down options, but I noticed while playing around with the software that an identical set of drop down options can be found under the "Change Node Type" drop down when right clicking on the side bare instead of on the whiteboard. Let me know if you want me to extend the same changes to that other drop down! I have attached a photo of what I am referring to.

![Screen Shot 2025-01-07 at 8 45 50 PM](https://github.com/user-attachments/assets/93ed9c05-daea-4c37-b58a-7eaf597ee83c)


-

keyserj reviewed (commented):
Nice work so far! I couldn't tell from your commit that you're new to open source & TypeScript 🙂 .

Re: the options under "Change node type": good catch, I forgot about that! I'd be happy to have you make the same changes there as well if you're up for it, otherwise I can create a separate ticket.

Also:

- the commit type `style` is intended for [_code_ style changes](https://github.com/amelioro/ameliorate/blob/32b6984befb27259f47c24ec35d5dfcb16c7592b/commitlint.config.cjs#L21); here `touchup` is more apt
- GitHub automatically links issues with syntax like `#629` - you've left the PR description with square brackets, so it isn't linked

rdubbi commented on README.md:
```diff
@@ -82,7 +82,9 @@ Thanks go to these wonderful people ([emoji key](https://allcontributors.org/doc
     <tr>
       <td align="center" valign="top" width="14.28%"><a href="https://seanyeh.com"><img src="https://avatars.githubusercontent.com/u/109418?v=4?s=100" width="100px;" alt="Sean Yeh"/><br /><sub><b>Sean Yeh</b></sub></a><br /><a href="https://github.com/amelioro/ameliorate/commits?author=seanyeh" title="Code">💻</a></td>
       <td align="center" valign="top" width="14.28%"><a href="https://shiv-iiitk.vercel.app/"><img src="https://avatars.githubusercontent.com/u/96898103?v=4?s=100" width="100px;" alt="Shiv Shankar "/><br /><sub><b>Shiv Shankar </b></sub></a><br /><a href="https://github.com/amelioro/ameliorate/commits?author=shibv" title="Code">💻</a></td>
+      <td align="center" valign="top" width="14.28%"><a href="https://www.linkedin.com/in/roy-badell/"><img src="https://avatars.githubusercontent.com/u/145716905?v=4" width="100px;" alt="Roy Badell"/><br /><sub><b>Roy Badell </b></sub></a><br /><a title="Code">💻</a><a title="Design">🎨</a></td>
```
Oh I would much rather have a bot do it! I will remove the change.

keyserj replied:
I actually hadn't included the `?s=100` in my suggestion, nor the update to `.all-contributorsrc`, so those still need to be done (or just remove the file and allow the bot to do it).

rdubbi replied:
I hit the commit suggestion button, thinking that that would allow the bot to do its magic. If this is not the case please let me know!

keyserj replied:
This is indeed not the case. I created the suggestion manually, and it only included the `href` changes I mentioned, not the `?s=100` parameter nor the update to `.all-contributorsrc`. It would probably be easiest to just remove your changes to this file from the PR, then we can have the bot make the changes in a follow-up PR that's automatically created after this one is merged.

rdubbi commented on src/web/common/components/ContextMenu/AddNodeMenuItem.tsx:
```diff
@@ -22,6 +23,9 @@ export const AddNodeMenuItem = ({ parentMenuOpen }: Props) => {
   // always showing breakdown + research nodes seems reasonable. justification nodes never need to be added on their own.
   const shownNodeTypes = breakdownNodeTypes.concat(researchNodeTypes);
 
+  // New Addition: Allows colors to be carried over from theme.ts. Roy 1/2025.
+  const theme = useTheme();
```
Good to know! I will clean those up.

rdubbi replied:
I am just now discovering the commit suggestion button!

rdubbi commented on src/web/common/components/ContextMenu/AddNodeMenuItem.tsx:
```diff
@@ -33,11 +37,26 @@ export const AddNodeMenuItem = ({ parentMenuOpen }: Props) => {
+              key={type}
+              onClick={() => addNodeWithoutParent(type, "diagram")}
+              sx={{
+                backgroundColor: theme.palette[type].main,
```
Personally, I think maintaining the consistency is more important than the risk of over-coloring in this instance. I also think coloring the icon may not work very well with options like mitigation, obstacle, solution, and question whose icons are made of thinner lines, and therefore may not have as much space on them to clearly display the color. 

From a UX/UI accessibility perspective, maintaining the background colors as they are now could be helpful for someone who has impaired eyesight/vision and is relying more heavily on the color than they are on the words or icons. 

All that being said... this is just my 2 cents.  You're in charge! Say the word, and I will make the change no problem!

keyserj replied:
You make a good point about thinner lines on some of the icons, making it inconsistent how visible the colors would be.

I think my comment may have been slightly confusing - I meant to suggest that coloring the _icon's_ background might be a good compromise between the over-coloring of coloring the entire _menu item's_ background and the under-coloring of coloring the icon _itself_. And that coloring the icon's background is more consistent, because it looks like the add node buttons that I imaged above. I don't think coloring the entire menu item's background is necessarily consistent with the node buttons. What do you think?

coloring the entire menu item's background:

![image](https://github.com/user-attachments/assets/84900248-444d-4f50-a1d3-1d09c64e3b8d)

coloring the icon itself:

![image](https://github.com/user-attachments/assets/c8d40d1d-d50c-475f-bda9-dcef8d074ebe)

coloring the icon's background:

![image](https://github.com/user-attachments/assets/23a92e66-629a-4450-8e47-df01e5ac0acb)

rdubbi replied:
I see! Very nice, I like it. Did you already make the changes yourself? Or is this just a visual you created somehow? 

Let me know if you need me to make the changes still. Otherwise, I hit resolve conversation.

keyserj replied:
I was trying out a few changes in the code, this is the one for the last image:
```tsx
            <ContextMenuItem key={type} onClick={() => addNodeWithoutParent(type, "diagram")}>
              <NodeIcon
                sx={{ backgroundColor: theme.palette[type].main }}
                className="mr-2 rounded p-0.5"
              />
              {title}
            </ContextMenuItem>
```

I'd still appreciate if you could get that into a commit and push it up.

rdubbi commented on src/web/common/components/ContextMenu/AddNodeMenuItem.tsx:
```diff
@@ -33,11 +37,26 @@ export const AddNodeMenuItem = ({ parentMenuOpen }: Props) => {
+                },
+              }}
+            >
+              <NodeIcon style={{ marginRight: 8 }} />
```
Makes sense. I am pretty new to Tailwind and using "style" is just closer to what I know. Thanks for sending these docs, I will give them a read!

rdubbi commented:
I would be happy to take on the "Change Node Types" task discussed. Please create a separate ticket and assign it to me if you can! 

Let me know if you want to swap background colors for icon colors too. 

I will be hiking and camping fri-sun and far far away from the computer, but I will check back first thing monday. 

Thanks for all your quick and insightful responses! I appreciate the guidance and patience.

keyserj commented:
> Please create a separate ticket and assign it to me if you can!

Created #636 - it seems that I can't assign you unless you comment on that issue though.

> I will be hiking and camping fri-sun and far far away from the computer, but I will check back first thing monday.

No rush! Have fun 🙂

keyserj commented on src/web/common/components/ContextMenu/AddNodeMenuItem.tsx:
```diff
@@ -33,11 +36,26 @@ export const AddNodeMenuItem = ({ parentMenuOpen }: Props) => {
-            {nodeDecorations[type].title}
-          </ContextMenuItem>
-        ))}
+        {/* New Addition: shownNodeTypes.map now includes icons and background color. Roy 1/2025. */}
```
This comment should be removed too, as mentioned here https://github.com/amelioro/ameliorate/pull/633#discussion_r1907274915. Generally I use GitHub's change suggestions to be explicit about changes, but they don't always include every change I imply by my comment.

rdubbi reviewed (commented):
Removed background color from whole menu item to only have background color for menu item icon, as discussed.

rdubbi commented:
It is my first time updating a PR in github after submission, I hope I did everything right! I changed the background colors as discussed and removed my manual contribution to allow the bot to redo it. I will wait for this to be merged before pulling down fresh code to start on #636 .

keyserj reviewed (approved):
Looks good! Some failing checks but it's ok:

- commitlint: failing because commits beyond the first don't match the required format; but I squash all commits on merge, so only the first one matters anyway
- lint: some of the code wasn't indented per the prettier standard - I just ran `npm run check-pretty -- --write` to fix this and pushed that up. (this should be running on file save, you may need setup for that described here https://github.com/amelioro/ameliorate/pull/580#discussion_r1864994128).
- e2e: this is a bug in my e2e check :sweat_smile: I'll fix this when I get the chance... but we can ignore for now

I'll merge this! Thanks for your contribution 🙂.

Before this change:

![finding-node-type](https://github.com/user-attachments/assets/87354f64-8aa5-478b-b483-b1cf9f641d9e)

keyserj commented:
@all-contributors please add @rdubbi for code and design
